src/pybind/mgr/dashboard/frontend/e2e/cluster/mgr-modules.po.ts

```diff
@@ -1,5 +1,616 @@
+    this.getTableCell('balancer').click();
+    element(by.cssContainingText('button', 'Edit')).click();
+
+    // Clears the editable field (pool_ids)
+    if (pool_ids[0] === undefined) {
+      element(by.id('pool_ids')).sendKeys(protractor.Key.chord(protractor.Key.CONTROL, 'a'));
+      element(by.id('pool_ids')).sendKeys(protractor.Key.BACK_SPACE);
+    }
+
+    // Checks that clearing represents in details tab of balancer
+    $('.btn.btn-secondary.tc_submitButton')
+      .click()
+      .then(() => {
+        this.navigateTo();
+        browser.wait(Helper.EC.visibilityOf(this.getTableCell('balancer')), 10000).then(() => {
+          this.getTableCell('balancer')
+            .click()
+            .then(() => {
+              if (pool_ids[0] !== undefined) {
+                browser.wait(
+                  Helper.EC.not(
+                    Helper.EC.textToBePresentInElement($$('.datatable-body').last(), 'balancer')
+                  ),
+                  Helper.TIMEOUT
+                );
+              }
+            });
+        });
+      });
+  }
+
+  editDashboard(api) {
+    // Selects the Dashboard manager module and edits the Audit_Api_Enabled to test if Dashboard has
+    // editing functionality. Then checks if it is reflected in the Details table. Only edits the
+    // Audit_Api_Enabled since there are many editable fields in the balancer. Could be added in a future PR.
+    this.navigateTo();
+    browser.wait(Helper.EC.elementToBeClickable(this.getTableCell('dashboard')), 10000);
+    this.getTableCell('dashboard').click();
+    element(by.cssContainingText('button', 'Edit')).click();
+
+    // clears api field and enters desired input
+    element(by.id('AUDIT_API_ENABLED')).sendKeys(protractor.Key.chord(protractor.Key.CONTROL, 'a'));
+    element(by.id('AUDIT_API_ENABLED')).sendKeys(protractor.Key.BACK_SPACE);
+    element(by.id('AUDIT_API_ENABLED')).sendKeys(api);
+
+    element(by.cssContainingText('button', 'Update'))
+      .click()
+      .then(() => {
+        this.navigateTo();
+        browser.wait(Helper.EC.visibilityOf(this.getTableCell('dashboard')), 10000).then(() => {
+          // Checks for visibility of dashboard in table
+          this.getTableCell('dashboard')
+            .click()
+            .then(() => {
+              browser.wait(
+                // checks if the api appears in details with the correct input
+                Helper.EC.textToBePresentInElement($$('.datatable-body').last(), api),
+                10000
+              );
+            });
+        });
+      });
+
+    // Clear Dashboard of edits
+    this.navigateTo();
+    browser.wait(Helper.EC.elementToBeClickable(this.getTableCell('dashboard')), 10000); // checks dashboard
+    this.getTableCell('dashboard').click();
+    element(by.cssContainingText('button', 'Edit')).click();
+
+    // Clears the editable field (api)
+    if (api !== undefined) {
+      element(by.id('AUDIT_API_ENABLED')).sendKeys(
```

Comment:
I think this 2 calls should be extracted to a `inputClear` method in Helper file